public/js/main.js

```diff
@@ -91,8 +90,9 @@ function renderToolbox() {
 }
 
 if (process.env.NODE_ENV !== "DEVTOOLS_PANEL") {
-  connectClient(tabs => {
+  connectClients().then((tabs) => {
     actions.newTabs(tabs);
+    renderNotConnected();
```

Comment:
It's confusing that you call `renderNotConnected` unconditionally; this feels like something that should be inside `renderToolbox`.

I just looked more at the code, and it took me a little bit to figure it out; `renderNotConnected` doesn't do anything if there are tabs. Personally I'd prefer to move that `isEmpty` check down here so that flow is more obvious and `renderNotConnected` only does 1 thing.